### PR TITLE
change epsilon in torch.nn.functional.normalize to avoid division by …

### PR DESCRIPTION
**Pull request by matthew-haines**
…0 with half precision. #32137
Demonstration (after fix):
image

### CODE REVIEW

**Comment by albanD** [comment]
I don't think we want to change the global default just to support half precision cc @gchanan 
This could lead to precision loss for most users.

Does passing a different epsilon when working with half precision work in your use case?

**Comment by vadimkantorov** [comment]
For my use case, option to pass a custom eps should be sufficient

The problem may stay if a default epsilon is used in some big library (e.g. some transformers) and then torch auto casting to fp16 is used. Then the user may not have control over the eps passed to normalize. For that case, maybe a generic hook that can replace nan/inf by zero (in outputs/grads) would be nice

**Comment by vadimkantorov** [comment]
```
eps = 1e-5
x = torch.tensor([0.0], dtype = torch.half, device = 'cuda).requires_grad_()
print(torch.autograd.grad(F.normalize(x, dim = -1, p = 2, eps = eps), (x,))[0])
```
`1e-5` seems to be too low, `1e-4` works

**Comment by vadimkantorov** [comment]
Should problems with epsilon / fp16 at least be documented in the docs? @ezyang
